lib/id3c/cli/command/redcap_sync.py

```diff
@@ -0,0 +1,315 @@
+                        + "identifiers above and all associated data from ID3C, run the following "
+                        + "command from within `id3c-production`:\n\n"
+                        + "```PGSERVICE=seattleflu-production envdir ~/workspace/seattleflu/env.d/redcap"
+                        + " pipenv run id3c redcap-sync delete "
+                        + f"{f'--api-url {project.api_url} --project-id {project.id} '}"
+                        + f"{f'--since-date {since_date} ' if since_date else ''}"
+                        + f"{f'--until-date {until_date} ' if until_date else ''}"
+                        + f"{f'--record-id {record_id} ' if record_id else ''}"
+                        + f"{f'--user {user} ' if user else ''}"
+                        + f"{f'--dag {dag} ' if dag else ''}"
+                        + f"{f'--return-format {return_format} ' if return_format else ''}"
+                        + "--commit```",
+                    },
+                },
+            ],
+        }
+        requests.post(
+            slack_url,
+            data=json.dumps(payload),
+            headers={"Content-type": "application/json"},
+            timeout=60,
+        )
+
+        LOG.debug("Record deletion message posted to Slack.")
+
+    deletion_count = 0
+    for encounter in del_encounters:
+        encounter_id = encounter.encounter_id
+        LOG.info(
+            f"Processing deleted encounter {encounter_id} from project {project.id}"
+        )
+
+        with db.savepoint(f"encounter {encounter.encounter_id}"):
+            delete_linked_encounter_records(db, encounter_id, encounter.individual_id)
+            deletion_count += 1
+
+        LOG.info(
+            f"Finished processing deleted encounter {encounter_id} from project {project.id}"
+        )
+
+    LOG.info(
+        f"Synced {len(deletion_events)} REDCap deletion events with ID3C. {deletion_count} ID3C encounters and associated data were removed."
+    )
+
+
+def delete_linked_encounter_records(
+    db: DatabaseSession, encounter_id: int, individual_id: int
+):
+    """
+    Deletes data associated with the linked `encounter_id` from the provided `db`.
+    If other encounter_ids are linked to the same record as the provided `encounter_id`,
+    don't delete those records.
+
+    This function may alter data from the following tables and should be used with care:
+    - `warehouse.encounter`
+    - `warehouse.encounter_location`
+    - `warehouse.individual`
+    - `warehouse.location`
+    - `warehouse.sample`
+    - `warehouse.presence_absence`
+    """
+    LOG.info(f"Deleting all relational encounter data for encounter {encounter_id}")
+
+    # Take a snapshot of relational location data before we delete our encounter.
+    location_ids_associated_with_this_encounter = db.fetch_all(
+        f"SELECT encounter_id, location_id FROM warehouse.encounter_location WHERE encounter_id = {encounter_id}"
+    )
```

Comment:
An alternative is to this and the other queries (that might simplify some logic below) is to select only the records that are safe to delete. e.g.
```
SELECT location_id FROM warehouse.encounter_location t1 WHERE encounter_id = {encounter_id}
AND location_id NOT IN (SELECT location_id FROM warehouse.encounter_location 
					   WHERE location_id = t1.location_id AND encounter_id <> {encounter_id})
```